core-addon/Core.js

```diff
@@ -580,13 +583,15 @@ module.exports = class Core {
    * is sent to the pipeline.
    *
    * @param {Object} data
-   *        A JSON-serializable object containing the demographics
+   *        A JSON-serializable obj0ect containing the demographics
    *        information submitted by the user.
    */
   async _updateDemographics(data) {
     await this._storage.setItem("demographicsData", data)
       .catch(e => console.error(`Core._updateDemographics - failed to save data`, e));
 
+    await this._sendStateUpdateToUI();
+
```

Comment:
We should rather do this in `_handleMessage`, similarly to how it's done for `"enrollment"`.